app/react/V2/Routes/Settings/CustomUploads/CustomUploads.tsx

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable func-call-spacing */
+/* eslint-disable no-spaced-func */
+/* eslint-disable max-statements */
+import React, { useEffect, useState } from 'react';
+import { LoaderFunction, useBlocker, useLoaderData, useRevalidator } from 'react-router-dom';
+import { IncomingHttpHeaders } from 'http';
+import { Row } from '@tanstack/react-table';
+import { useSetRecoilState } from 'recoil';
+import { Translate } from 'app/I18N';
+import { FetchResponseError } from 'shared/JSONRequest';
+import { FileType } from 'shared/types/fileType';
+import { getByType, remove, UploadService } from 'V2/api/files';
+import { Button, ConfirmationModal, Table } from 'V2/Components/UI';
+import { SettingsContent } from 'V2/Components/Layouts/SettingsContent';
+import { ConfirmNavigationModal } from 'V2/Components/Forms';
+import { notificationAtom } from 'V2/atoms';
+import {
+  createColumns,
+  FileList,
+  UploadProgress,
+  DropzoneModal,
+  EditFileSidepanel,
+} from './components';
+
+const customUploadsLoader =
+  (headers?: IncomingHttpHeaders): LoaderFunction<FileType[]> =>
+  async () => {
+    const files = await getByType('custom', headers);
+    return files;
+  };
+
+const uploadService = new UploadService('custom');
+
+const CustomUploads = () => {
+  const files = useLoaderData() as FileType[];
+  const setNotifications = useSetRecoilState(notificationAtom);
+  const revalidator = useRevalidator();
+  const [selectedRows, setSelectedRows] = useState<Row<FileType>[]>([]);
+  const [fileToEdit, setFileToEdit] = useState<FileType>();
+  const [confirmationModal, setConfirmationModal] = useState(false);
+  const [showUploadsModal, setShowUploadsModal] = useState(false);
+  const [confirmNavigationModal, setConfirmNavigationModal] = useState(false);
+  const [showSidepanel, setShowSipanel] = useState(false);
+  const [modalProps, setModalProps] = useState<{
+    action: () => void;
+    items: Row<FileType>[] | FileType[];
+  }>({
+    action: () => {},
+    items: [],
+  });
+
+  const blocker = useBlocker(Boolean(uploadService.isUploading()));
+
+  useEffect(() => {
+    if (blocker.state === 'blocked') {
+      setConfirmNavigationModal(true);
+    }
+
+    if (blocker.state === 'proceeding') {
+      uploadService.abort();
+    }
+  }, [blocker, setConfirmNavigationModal]);
+
+  const notify = (responses: (FileType | FetchResponseError)[], message: React.ReactNode) => {
+    const hasErrors = responses.find(response => response instanceof FetchResponseError);
```

Comment:
I don't think we should delegate the show or not show an error based on a type, that can change in the future.
This needs to be a try-catch or similar. If you go to the route and throw an error on the request this does not show anything.

Reply:
I can also check if there's at least a valid file in the response to determine if there's an error. But I believe we should look into how we want to detect errors in responses generally, since other places in the new routes are doing this.